flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/source/MySqlParallelSourceTestBase.java

```diff
@@ -79,277 +63,58 @@
-                    "+I[2003, user_24, Shanghai, 123567891234]",
-                };
-        List<String> expectedBinlogData = new ArrayList<>();
-        for (int i = 0; i < captureCustomerTables.length; i++) {
-            expectedBinlogData.addAll(Arrays.asList(binlogForSingleTable));
-        }
-        String[] expectedBinlog = expectedBinlogData.toArray(new String[0]);
-        assertThat(fetchRows(iterator, expectedBinlog.length), containsInAnyOrder(expectedBinlog));
-        tableResult.getJobClient().get().cancel().get();
-    }
-
-    private static List<String> fetchRows(Iterator<Row> iter, int size) {
-        List<String> rows = new ArrayList<>(size);
-        while (size > 0 && iter.hasNext()) {
-            Row row = iter.next();
-            rows.add(row.toString());
-            size--;
-        }
-        return rows;
-    }
-
-    private String getTableName(String[] captureCustomerTables) {
-        if (captureCustomerTables.length == 1) {
-            return "customers";
-        } else {
-            // pattern that matches test table: customers and customers_1
-            return "customers.*";
-        }
-    }
-
-    private String getServerId() {
-        final Random random = new Random();
-        int serverId = random.nextInt(100) + 5400;
-        return serverId + "-" + (serverId + PARALLELISM);
-    }
-
-    private void sleepMs(long millis) {
+    private static String buildMyCnfThatEnableGtid() {
         try {
-            Thread.sleep(millis);
-        } catch (InterruptedException ignored) {
+            File resourceFolder =
+                    Paths.get(
+                                    Objects.requireNonNull(
+                                                    MySqlParallelSourceTestBase.class
+                                                            .getClassLoader()
+                                                            .getResource("."))
+                                            .toURI())
+                            .toFile();
+
+            TemporaryFolder tempFolder = new TemporaryFolder(resourceFolder);
+            tempFolder.create();
+            File folder = tempFolder.newFolder(String.valueOf(UUID.randomUUID()));
+            Path cnf = Files.createFile(Paths.get(folder.getPath(), "my.cnf"));
+
+            String mysqldConf =
+                    "[mysqld]\n"
+                            + "skip-host-cache\n"
+                            + "skip-name-resolve\n"
+                            + "user=mysql\n"
+                            + "symbolic-links=0\n"
+                            + "binlog_format = row\n"
+                            + "log_bin = mysql-bin\n"
+                            + "server-id = 223344\n"
+                            + "binlog_row_image = FULL\n";
+
+            String gitModeSetting =
+                    "# enable gitid mode\n"
+                            + "gtid_mode = on\n"
+                            + "enforce_gtid_consistency = on\n";
```

Comment:
Why not add a separate `my_gtid.conf` file which I think it more easier to maintain ?

Reply:
I improve it, the reason is that TestContainer only recognize the conf name `my.cnf`, `my_gtid.conf` will be ignored

Reply:
OK. Then we can put `my.conf` in a `mysql-gtids` folder.